syntax = "proto3";
package ups;
enum Phase {
    SINGLE_PHASE=0;
    THREE_PHASE=1;
}
enum LOAD{
	LINEAR=0;
	NON_LINEAR=1;
}
enum MODE{
	NORMAL_MODE=0;
	STORAGE_MODE=1;
	FAULT_MODE=2;
	ALARM_MODE=4;
}

message OverLoad{
    uint32 load_percentage=1;
    uint32 overload_time_min=2;
}
message spec{
	Phase phase=1;
	uint32 rated_va = 2;
	uint32 rated_voltage = 3;
	uint32 rated_current =4;
	uint32  min_input_voltage = 5;
	uint32  max_input_voltage = 6;
	uint32 pf_rated_current=7;
	uint32 max_continous_amp=8;
	uint32 overload_amp=9;
    OverLoad overload_long=10;
    OverLoad overload_medium=11;
    OverLoad overload_short=12;
	uint64 avg_switch_time_ms = 20;
	uint64 avg_backup_time_ms = 21;
}